stellar-contract-env-host/src/host.rs

```diff
@@ -1432,7 +1432,54 @@ impl CheckedEnv for Host {
         j: RawVal,
         l: RawVal,
     ) -> Result<RawVal, HostError> {
-        todo!()
+        // offset in the host binary
+        let i: usize = u32::try_from(i).map_err(|_| {
+            HostError::WithStatus(
+                String::from("i must be u32"),
+                ScStatus::HostFunctionError(ScHostFnErrorCode::InputArgsWrongType),
+            )
+        })? as usize;
```

Comment:
I see what you saying. Though I still feel we should be able to make the assumption that usize is at least 32-bits (i.e. no one will run the host crate in a 16-bit microcontroller). Otherwise, almost every other design decision in the host will fail, such as `RawVal` types.  I think there should be a top-level, global check that ensures that `usize.BITS >= 32` (or a similar check). cc @graydon